File: backend/src/main/java/com/sentinel/backend/entity/LancamentoFinanceiro.java
package com.sentinel.backend.entity;

import java.math.BigDecimal;
import java.util.Date;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "lancamentos_financeiros")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class LancamentoFinanceiro {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Date data;
    private BigDecimal saldoDiario;
    private BigDecimal saldoMensal;
}

File: frontend/src/app/models/lancamento-financeiro.ts
export class LancamentoFinanceiro {
  id?: number;
  data: Date | null;
  saldoDiario: number;
  saldoMensal: number;

  constructor(
    data: Date | null,
    saldoDiario: number,
    saldoMensal: number,
    id?: number
  ) {
    this.id = id;
    this.data = data;
    this.saldoDiario = saldoDiario;
    this.saldoMensal = saldoMensal;
  }
}
